src/fielddef.ts

```diff
@@ -251,6 +251,7 @@ export function defaultType(fieldDef: FieldDef, channel: Channel): Type {
  */
 export function normalize(fieldDef: ChannelDef, channel: Channel) {
   // If a fieldDef contains a field, we need type.
+  fieldDef = JSON.parse(JSON.stringify(fieldDef)); // Clone
```

Comment:
@kanitw I couldn't figure out a good way to remove side effects without cloning it. If you think there's one, I'll update it